apps/src/applab/gridUtils.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * A couple of utility functions for dealing with our design mode grid.
+ */
+module.exports = {
+  GRID_SIZE: 5,
+  /**
+   * Given an element being dragged, determine the scaled x/y position of the
+   * top left corned, scaled to our visualization.
+   * @param {jQueryObject} draggingElement
+   * @return {object}
+   */
+  scaledDropPoint: function (draggingElement) {
+    var div = document.getElementById('designModeViz');
+
+    var xScale = div.getBoundingClientRect().width / div.offsetWidth;
+    var yScale = div.getBoundingClientRect().height / div.offsetHeight;
+
+    var left = (draggingElement.offset().left - $(div).offset().left) / xScale;
+    var top = (draggingElement.offset().top - $(div).offset().top) / yScale;
+
+    // snap top-left corner to nearest location in the grid
+    left = this.snapToGridSize(left);
+    top = this.snapToGridSize(top);
```

Comment:
For example, if we have access to the cached transform, I think this simplifies to:
```javascript
// Assuming draggedCorner is a cached SVGPoint, because they're a little expensive to create
var offset = draggingElement.offset();
draggedCorner.x = offset.left;
draggedCorner.y = offset.top;
draggedCorner = draggedCorner.matrixTransform(screenSpaceToAppSpaceTransform);
draggedCorner.x = this.snapToGridSize(draggedCorner.x);
draggedCorner.y = this.snapToGridSize(draggedCorner.y);
```
Of course, that means exposing the cached transform and related work in a way both the vizoverlay and designmode can use.  What if instead of a module, the grid utils became a ApplabGrid class that we instantiated, which stored the current scale, transform, and utility methods for interacting with the grid?  Then we injected the ApplabGrid into both the VisualizationOverlay and DesignModes, to use as they need it.